File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
    vector<int> ans;
    for (int i = 0; i < numbers.size(); i++) {
        int low = i + 1;
        int high = numbers.size() - 1;
        while (low <= high) {
            int mid = low + (high - low) / 2;
            if (numbers[i] + numbers[mid] == target) {
                ans.push_back(i + 1);
                ans.push_back(mid + 1);
                return ans;}
             else if (numbers[i] + numbers[mid] > target) 
              {high = mid - 1;}
              else{low = mid + 1;}
        }
    }
    return ans;
}
};

File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        lo = 0
        hi = len(numbers)-1
        while lo<=hi:
            chk = numbers[lo]+numbers[hi]

            if chk ==target :
                return [lo+1,hi+1]
            elif chk>target:
                hi-=1
            else:
                lo+=1
        return [-1,-1]